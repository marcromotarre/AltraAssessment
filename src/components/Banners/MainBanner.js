import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import BannerSmallSVG from './../../svg/banner-mini.svg';
import FilterSVG from './../../svg/filter.svg';

export const BannerContainer = styled.div`
  position: relative;
  background: white;
  width: 100%;
  height: 72px;
  display: grid;
  grid-template-columns: 15% 35% 35% 15%;
  grid-template-areas: '. m-town-name m-town-name m-filter';

  label {
    grid-area: m-town-name;
    font-family: 'HelveticaNeue-Thin';
    font-weight: 300;
    font-size: 42px;
    align-self: center;
    justify-self: center;
  }
`;

const BannerBackground = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  svg {
    width: 90%;
    opacity: 0.17;
  }
`;

const BannerSmall = styled(BannerSmallSVG)`
  width: 90%;
  opacity: 0.17;
  position: absolute;
`;

const FilterButton = styled.button`
  grid-area: m-filter;
  align-self: center;
  justify-self: center;
  z-index: 1;
`;

const MainBanner = ({ townName, onClickFilterIcon, showFilterButton }) => {
  const handleClickFilter = () => {
    onClickFilterIcon();
  };

  return (
    <BannerContainer>
      <BannerBackground>
        <BannerSmall />
      </BannerBackground>
      {townName && <label>{townName}</label>}
      {showFilterButton && (
        <FilterButton onClick={handleClickFilter}>
          <FilterSVG width={21.5} />
        </FilterButton>
      )}
    </BannerContainer>
  );
};

MainBanner.propTypes = {
  townName: PropTypes.string,
  onClickFilter: PropTypes.func,
  showFilterButton: PropTypes.bool,
};

MainBanner.defaultProps = {
  townName: '',
  onClickFilter: () => {},
  showFilterButton: false,
};

export default MainBanner;
