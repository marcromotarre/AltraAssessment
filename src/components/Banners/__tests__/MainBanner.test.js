import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MainBanner from '../MainBanner';

/** this component should:
 *  - render Main Banner with:
 *      - svg background (styled.div)
 *          - svg background icon (styled.svg)
 *      - filter icon (styled.svg)
 *      - town name (label) (optional --> only if props)
 */

const props = {
  townName: 'town name',
  showFilterButton: true,
  onClickFilter: jest.fn(),
};

const noTownNameProps = {
  showFilterButton: true,
  onClickFilter: jest.fn(),
};

const noShowFilterButtonProps = {
  townName: 'town name',
  onClickFilter: jest.fn(),
};

describe('Main Banner Component tests', () => {
  it('MainBanner Snapshot', () => {
    const tree = renderer.create(<MainBanner {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('MainBanner renders BackgroundIcon, TownName and Filter Icon', () => {
    const wrapper = shallow(<MainBanner {...props} />);

    expect(wrapper.children().length).toBe(3);
    const bannerBackground = wrapper.childAt(0);
    expect(bannerBackground.type().displayName).toBe('styled.div');
    expect(bannerBackground.children().length).toBe(1);
    const bannerSmallIcon = bannerBackground.childAt(0);
    expect(bannerSmallIcon.type().displayName).toBe('styled.svg');

    const townNameLabel = wrapper.childAt(1);
    expect(townNameLabel.type()).toBe('label');
    expect(townNameLabel.text()).toBe(props.townName);

    const filterButton = wrapper.childAt(2);
    const filterIcon = filterButton.childAt(0);
    expect(filterButton.type().displayName).toBe('styled.button');
    expect(filterIcon.type()).toBe('svg');
    expect(filterIcon.prop('width')).toBe(21.5);
  });

  it('should not render TownName Label if no townName in props', () => {
    const wrapper = shallow(<MainBanner {...noTownNameProps} />);
    expect(wrapper.children().length).toBe(2);
    const bannerBackground = wrapper.childAt(0);
    expect(bannerBackground.type().displayName).toBe('styled.div');
    const filterButton = wrapper.childAt(1);
    expect(filterButton.type().displayName).toBe('styled.button');
  });

  it('should not render FilterIcon Button if props showFilterButton is false', () => {
    const wrapper = shallow(<MainBanner {...noShowFilterButtonProps} />);
    expect(wrapper.children().length).toBe(2);
    const bannerBackground = wrapper.childAt(0);
    expect(bannerBackground.type().displayName).toBe('styled.div');
    const townNameLabel = wrapper.childAt(1);
    expect(townNameLabel.type()).toBe('label');
  });
});
