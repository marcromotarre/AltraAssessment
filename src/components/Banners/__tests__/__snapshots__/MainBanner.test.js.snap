// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Banner Component tests MainBanner Snapshot 1`] = `
<div
  className="sc-AxjAm dCtgNG"
>
  <div
    className="sc-AxirZ bnKdnV"
  >
    <svg
      className="sc-AxiKw iKXCa"
    />
  </div>
  <label>
    town name
  </label>
  <button
    className="sc-AxhCb bjuFrh"
    onClick={[Function]}
  >
    <svg
      width={21.5}
    />
  </button>
</div>
`;
